doc/design/draft/named-segments.md

```diff
@@ -0,0 +1,228 @@
+must rely on configuring differently sized memory pools to support topic data of different sizes. 
+This has a number of drawbacks, including:
+
+* Processes communicating over otherwise independent topics share the same memory pool. 
+If any one process loans too many chunks, or fails to release chunks, it can cause starvation of ALL processes 
+relying on this memory pool.
+* Topic data allocation is assigned to memory pools via "bucketing" - or finding the memory pool with the smallest 
+chunk size to support the loan request. This means that even if one attempts to cater a memory pool to a specific topic,
+any other topics with the same size requirement will be forced to use the same memory pool.
+* Faulty publishers which e.g. write past the end of their loaned message may cause faults in any process communicating
+over the same shared memory segment. These faults will be difficult to trace, especially because the faulty publisher
+may never write past the end of the mapped shared memory segment, and therefore may never trigger a segfault.
+
+The above points demonstrate how the current design violates the principal of [Freedom From Interference](https://heicon-ulm.de/en/iso26262-freedom-from-interference-what-is-that/).
+
+### Proposal
+
+In order to overcome the above limitations, shared memory segments shall be mapped by 
+name instead of by access control. This will increase flexibility while still granting all the benefits of the current access control model.
+
+## Requirements
+
+* The RouDi configuration must support specifying names for shared memory segments.
+* If a name is not configured, the fallback behavior must be the same as it is currently - assigning the name of the current process user.
+* Communication endpoints should be able to specify which segments they intend to use.
+  * Publishers may specify a single segment into which they will publish.
+  * Subscribers may specify one or more segments from which they may receive messages.
+  * Clients and Servers may specify a single segment into which they send requests/responses, and multiple segments from which they receive responses/requests.
+* When no segment is specified, the fallback behavior must be the same as it is currently
+  * Publishers will select the *unique* segment they have write access to.
+  * Subscribers will receive messages from all segments they have read access to.
+  * Clients and Servers follow the same rules as Publishers for data they send and the same rules as Subscribers for data they receive.
+* Creation will fail if a communication endpoint requests to use a segment to which it does not
+have the proper access rights.
+
+### Optional additional requirements
+
+* When initializing the runtime, a list of segment names may be provided. In this case, rather than mapping all shared memory segments available in the system, only those that are named shall be mapped.
```

Comment:
When we create a new API, the user should explicitely tell what the behavior should be by either passing a list of segments or using a flag which indicates to get all segments with read or write access. The current API will keep its behavior but would be deprecated at the long term.

Reply:
This seems reasonable but it would be a somewhat breaking API change as users would need to include the flag, or explicitly pass an empty list that they are not currently passing. 

This is in contrast to simply not providing a list, which can be interpreted as the deprecated behavior. I'm a bit ambivalent here. I'm open to a small initial breaking change followed by dropping support for the old behavior. Or we could simply start by printing a warning when no list/flag is provided.

Reply:
I think I  need some pseudo code for this :sweat_smile:

It also feels a bit complicated with the implicit behavior when something is omitted.